swift-sdk/Internal/InAppManager+Functions.swift

```diff
@@ -122,20 +122,34 @@ struct MessagesObtainedHandler {
         let removedInboxCount = removedMessages.reduce(0) { $1.saveToInbox ? $0 + 1 : $0 }
         let addedInboxCount = addedMessages.reduce(0) { $1.saveToInbox ? $0 + 1 : $0 }
         
+        var messagesOverwritten = 0
         var newMessagesMap = OrderedDictionary<String, IterableInAppMessage>()
-        messages.forEach {
-            if let existingMessage = messagesMap[$0.messageId] {
-                newMessagesMap[$0.messageId] = existingMessage
+        messages.forEach { serverMessage in
+            let messageId = serverMessage.messageId
+            if let existingMessage = messagesMap[messageId] {
+                if Self.shouldOverwrite(clientMessage: existingMessage, withServerMessage: serverMessage) {
+                    newMessagesMap[messageId] = serverMessage
+                    messagesOverwritten += 1
+                } else {
+                    newMessagesMap[messageId] = existingMessage
+                }
             } else {
-                newMessagesMap[$0.messageId] = $0
+                newMessagesMap[messageId] = serverMessage
             }
         }
         
-        return MergeMessagesResult(inboxChanged: removedInboxCount + addedInboxCount > 0,
+        return MergeMessagesResult(inboxChanged: removedInboxCount + addedInboxCount + messagesOverwritten > 0,
```

Comment:
We should also add a test for this.

Reply:
It's now covered with the test in `InAppMessageProcessorTests`